<?xml version="1.0"?>
<sdf version="1.5">
  <world name="maze">

    <physics name="default_physics" default="true" type="ode">
      <gravity>0 0 -9.8</gravity>
    </physics>

    <include>
      <name>sun</name>
      <uri>model://sun</uri>
    </include>

    <light name="point_light" type="point">
      <pose>0 0 1 0 0 0</pose>
      <diffuse>1 1 1 1</diffuse>
      <specular>0.1 0.1 0.1 1.0</specular>
      <attenuation>
        <range>10</range>
        <constant>0.5</constant>
        <linear>0.5</linear>
        <quadratic>0</quadratic>
      </attenuation>
      <cast_shadows>false</cast_shadows>
      <direction>0 0 -1</direction>
    </light>

    <include>
      <name>sim_world</name>
      <uri>model://maze</uri>
      <static>true</static>
      <pose>0 0 0 0 0 0</pose>
    </include>

    <!-- -->
    <!-- test_ball1（Blue） -->
    <!-- -->
    <model name="test_ball1">
      <pose>-0.4 0.0 1 0 0 0</pose>
      <link name="ball_link1">
        <inertial>
          <mass>0.035</mass>
          <inertia>
            <ixx>0.0001</ixx>
            <iyy>0.0001</iyy>
            <izz>0.0001</izz>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyz>0</iyz>
          </inertia>
        </inertial>
	<!-- 滑り＆跳ね防止 -->
        <velocity_decay>
          <linear>0.0</linear>
          <angular>0.2</angular>
        </velocity_decay>
	<collision name="collision">
          <geometry>
            <sphere><radius>0.033</radius></sphere>
          </geometry>
          <surface>
            <friction>
              <ode>
                <mu>0.5</mu>
		<mu2>0.5</mu2>
              </ode>
            </friction>
          </surface>
        </collision>
        <visual name="visual">
          <geometry>
            <sphere><radius>0.033</radius></sphere>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Blue</name>
            </script>
          </material>
        </visual>
      </link>
    </model>

    <!-- -->
    <!-- test_ball2（Red） -->
    <!-- -->
    <model name="test_ball2">
      <pose>-0.6 -0.1 1 0 0 0</pose>
      <link name="ball_link2">
        <inertial>
          <mass>0.035</mass>
          <inertia>
            <ixx>0.0001</ixx>
            <iyy>0.0001</iyy>
            <izz>0.0001</izz>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyz>0</iyz>
          </inertia>
        </inertial>
	<!-- 滑り＆跳ね防止 -->
        <velocity_decay>
          <linear>0.0</linear>
          <angular>0.2</angular>
        </velocity_decay>
	<collision name="collision">
          <geometry>
            <sphere><radius>0.033</radius></sphere>
          </geometry>
          <surface>
            <friction>
              <ode>
                <mu>0.5</mu>
                <mu2>0.5</mu2>
              </ode>
            </friction>
          </surface>
        </collision>
        <visual name="visual">
          <geometry>
            <sphere><radius>0.033</radius></sphere>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Red</name>
            </script>
          </material>
        </visual>
      </link>
    </model>

    <!-- -->
    <!-- test_ball3（Yellow） -->
    <!-- -->
    <model name="test_ball3">
      <pose>-0.6 0.1 1 0 0 0</pose>
      <link name="ball_link3">
        <inertial>
          <mass>0.035</mass>
          <inertia>
            <ixx>0.0001</ixx>
            <iyy>0.0001</iyy>
            <izz>0.0001</izz>
            <ixy>0</ixy>
            <ixz>0</ixz>
            <iyz>0</iyz>
          </inertia>
        </inertial>
	<!-- 滑り＆跳ね防止 -->
        <velocity_decay>
          <linear>0.0</linear>
          <angular>0.2</angular>
        </velocity_decay>
	<collision name="collision">
          <geometry>
            <sphere><radius>0.033</radius></sphere>
          </geometry>
          <surface>
            <friction>
              <ode>
                <mu>0.5</mu>
                <mu2>0.5</mu2>
              </ode>
            </friction>
          </surface>
        </collision>
        <visual name="visual">
          <geometry>
            <sphere><radius>0.033</radius></sphere>
          </geometry>
          <material>
            <script>
              <uri>file://media/materials/scripts/gazebo.material</uri>
              <name>Gazebo/Yellow</name>
            </script>
          </material>
        </visual>
      </link>
    </model>

  </world>
</sdf>

